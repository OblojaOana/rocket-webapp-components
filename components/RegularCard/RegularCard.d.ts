import { CardProps } from '@material-ui/core/Card';
import { CardContentProps } from '@material-ui/core/CardContent';
import { CardHeaderProps } from '@material-ui/core/CardHeader';

export interface Props {

    /**
     * Override or extend the styles applied to the card
     */
    customCardClasses?: string;

    /**
     * Override or extend the styles applied to the title of the card
     */
    customCardTitleClasses?: string;

    /**
     * If set to true, the background will be transparent
     */
    plainCard?: boolean;

    /**
     * The title of the card
     */
    cardTitle?: React.ReactNode;

    /**
     * The subtitle of the card
     */
    cardSubtitle?: React.ReactNode;

    /**
     * The content of the card
     */
    content?: React.ReactNode;

    /**
     * Title alignment
     */
    titleAlign?: "right" |
    "left" |
    "center"

    /**
     * Content alignment
     */
    contentAlign?: "right" |
    "left" |
    "center"

    /**
     * Subtitle alignment
     */
    subtitleAlign?: "right" |
    "left" |
    "center"

    /**
     * Demos:
     * 
     * - https://bit.dev/totalsoft_oss/react-mui/regular-card
     * 
     *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/develop/components/RegularCard/RegularCard.js
     */

}

export type RegularCardProps = Props | CardProps | CardContentProps | CardHeaderProps

export default function RegularCard(props: RegularCardProps): JSX.Element;