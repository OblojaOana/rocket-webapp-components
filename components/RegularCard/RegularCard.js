import React from "react";
import { Card, CardContent, CardHeader, makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";
import cx from "classnames";

import regularCardStyle from "./regularCardStyle";
const useStyles = makeStyles(regularCardStyle);

const RegularCard = ({
  plainCard,
  cardTitle,
  cardSubtitle,
  content,
  titleAlign,
  customCardClasses,
  contentAlign,
  subtitleAlign,
  customCardTitleClasses
}) => {
  const classes = useStyles();

  const cardClasses =
    classes.card +
    cx({
      [" " + classes.cardPlain]: plainCard,
      [" " + customCardClasses]: customCardClasses !== undefined
    });
  return (
    <Card className={cardClasses}>
      {cardTitle !== undefined || cardSubtitle !== undefined ? (
        <CardHeader
          classes={{
            root: classes.cardHeader,
            title:
              classes.cardTitle +
              " " +
              classes[titleAlign] +
              " " +
              customCardTitleClasses,
            subheader: classes.cardSubtitle + " " + classes[subtitleAlign]
          }}
          title={cardTitle}
          subheader={cardSubtitle}
        />
      ) : null}
      <CardContent
        className={classes.cardContent + " " + classes[contentAlign]}
      >
        {content}
      </CardContent>
    </Card>
  );
};

RegularCard.propTypes = {
  customCardClasses: PropTypes.string,
  customCardTitleClasses: PropTypes.string,
  plainCard: PropTypes.bool,
  cardTitle: PropTypes.node,
  cardSubtitle: PropTypes.node,
  content: PropTypes.node,
  titleAlign: PropTypes.oneOf(["right", "left", "center"]),
  contentAlign: PropTypes.oneOf(["right", "left", "center"]),
  subtitleAlign: PropTypes.oneOf(["right", "left", "center"])
};

export default RegularCard;
