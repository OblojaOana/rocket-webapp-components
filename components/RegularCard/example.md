##### Copy this snapshot in the overview panel

```js
import React from "react";
import RegularCard from "@bit/totalsoft_oss.react-mui.regular-card";

export default (
  <RegularCard
    cardTitle={"Title"}
    cardSubtitle={"Subtitle"}
    content={<div>Content</div>}
  />
);
```
