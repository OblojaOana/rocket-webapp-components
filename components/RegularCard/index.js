import RegularCard from "../../core/RegularCard";
export default RegularCard;
