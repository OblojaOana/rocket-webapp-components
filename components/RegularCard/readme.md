##### Copy this snapshot in the overview panel

```js
import React from "react";
import Pagination from "@bit/totalsoft_oss.react-mui.pagination";

export default (
  <RegularCard
    cardTitle={"Title"}
    cardSubtitle={"Subtitle"}
    content={<div>Content</div>}
  />
);
```
