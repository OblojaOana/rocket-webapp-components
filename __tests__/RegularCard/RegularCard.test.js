import React from "react";
import renderer from "react-test-renderer";
import RegularCard from "../../components/RegularCard/RegularCard";

describe("Render RegularCard", () => {
  it("render correctly RegularCard component", () => {
    const component = renderer.create(
      <RegularCard
        cardTitle={"Title"}
        cardSubtitle={"Subtitle"}
        content={"Card content"}
        titleAlign="center"
        contentAlign="center"
        subtitleAlign="center"
      />
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
