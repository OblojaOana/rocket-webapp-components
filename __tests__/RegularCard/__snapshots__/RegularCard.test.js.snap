// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegularCard Snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root makeStyles-cardHeader-3"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title makeStyles-cardTitle-4 makeStyles-center-7 undefined MuiTypography-h5 MuiTypography-displayBlock"
      >
        Title
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader makeStyles-cardSubtitle-8 makeStyles-center-7 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Subtitle
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root makeStyles-cardContent-9 makeStyles-center-7"
  >
    Card content
  </div>
</div>
`;
